### PR TITLE
Change license to BSL-1.1

### PR DESCRIPTION
**Pull request by rijkvanzanten**
Today Directus is adopting BSL 1.1 — with additional usage grants added in to keep the platform open and available. For nearly all of our community, today’s announcement won’t change a thing, but licenses can be intimidating, so let’s highlight the key points below:

- All Directus source code will still be open and available on GitHub
- Non-production use of Directus is still completely free for everyone
- Production use of Directus is still completely free for nearly all users
- Everyone can still use the entire Directus platform for free… unless you (your legal entity) exceed $5,000,000 USD in “total finances” per year
- Code released under this new license converts to GPLv3 (OSS) after 3 years

Massive thanks to everybody who chimed in on the discussion in https://github.com/directus/directus/discussions/17977 and on [Discord](https://directus.chat/) ❤️ 

For more information, please see our [blog post](https://directus.io/blog/why-we-are-relicensing-directus/) and [FAQ](https://directus.io/bsl-faq).

List of all the license updates below:

| Package                               | New License |
| ------------------------------------- | ----------- |
| `directus`                            | BSL 1.1     |
| `@directus/api`                       | BSL 1.1     |
| `@directus/app`                       | BSL 1.1     |
| `@directus/composables`               | MIT         |
| `@directus/constants`                 | MIT         |
| `@directus/create-directus-extension` | MIT         |
| `@directus/exceptions`                | MIT         |
| `@directus/extensions-sdk`            | MIT         |
| `@directus/schema`                    | BSL 1.1     |
| `@directus/specs`                     | BSL 1.1     |
| `@directus/storage`                   | BSL 1.1     |
| `@directus/storage-driver-azure`      | BSL 1.1     |
| `@directus/storage-driver-cloudinary` | BSL 1.1     |
| `@directus/storage-driver-gcs`        | BSL 1.1     |
| `@directus/storage-driver-local`      | BSL 1.1     |
| `@directus/storage-driver-s3`         | BSL 1.1     |
| `@directus/types`                     | MIT         |
| `@directus/update-check`              | MIT         |
| `@directus/utils`                     | MIT         |

### CODE REVIEW

**Comment by w0kyj** [comment]
Should the referenced blog and faq be links to those locations?

**Comment by rijkvanzanten** [comment]
> Should the referenced blog and faq be links to those locations?

Yeah once they're live I'll update the description 👍🏻

**Review by licitdev — COMMENTED** [review]
Do we need a license for the `docs` package?

What should the license be in the OpenAPI specs? The version should be updated alongside as well.

https://github.com/directus/directus/blob/ca64c6fce523f120d423f2c138b615b90258eb51/packages/specs/src/openapi.yaml#L5-L10

Additionally, I think we should include in the MIT license file for packages under MIT license.

**Comment by rijkvanzanten** [comment]
> Do we need a license for the docs package?

No, that's not published anywhere nor intended for re-use. 

> What should the license be in the OpenAPI specs? The version should be updated alongside as well.

Ah good one. I searched for the smaller string, let me fix that. 

> Additionally, I think we should include in the MIT license file for packages under MIT license.

I did! 🙂

**Review by azrikahar — COMMENTED** [review]
I believe these changes: https://github.com/directus/directus/pull/18330/files#diff-5a831ea67cf5cf8703b0de46901ab25bd191f56b320053be9332d9a3b0d01d15L54-R59

will need to be reflected in these places as well:

- [x] https://github.com/directus/directus/blob/381f1c9d8e6c1c799a19f34e12cbde88d525324a/directus/readme.md?plain=1#L86-L89
- [x] https://github.com/directus/directus/blob/381f1c9d8e6c1c799a19f34e12cbde88d525324a/packages/specs/README.md?plain=1#L41-L44

(or maybe we can remove those section since other packages' readme file doesn't have that section)

**Comment by licitdev** [comment]
I did a search for `MIT` and the license did not show up. 🤦  Could we add a header to include the words `MIT License` before the copyright line?

Ref: https://choosealicense.com/licenses/mit/

File: packages/specs/src/openapi.yaml [hunk 1]
```diff
@@ -5,9 +5,9 @@ info:
   contact:
     email: contact@directus.io
   license:
-    name: GPL-3.0
-    url: 'https://www.gnu.org/licenses/gpl-3.0.html'
-  version: 9.0.0
+    name: BUSL-1.1
+    url: 'https://directus.io/bsl'
+  version: 10
```
Comment (paescuj) [inline packages/specs/src/openapi.yaml:10]:
```suggestion
  version: 10.0.0
```

**Review by licitdev — APPROVED** [review]

**Review by azrikahar — APPROVED** [review]

**Review by paescuj — APPROVED** [review]
LGTM... and double-checked all packages for the right license 🚀🐰 

giphy

**Review by benhaynes — APPROVED** [review]
Yeehaw!

**Comment by koksikus** [comment]
What does this entry mean? "Change Date: Three years from release date"
I know that the law is not retroactive.

**Comment by rijkvanzanten** [comment]
> So in 3 years from today version 10 will be available for free? 🤔 I don't understand this - I'm not an expert and I'm asking out of pure curiosity

That's correct! 3 years from a release, the release itself is officially back to GPL-3 🙂

**Comment by koksikus** [comment]
Interesting point. 🤔 From the technical side, version 10.0.0 after 3 years will no longer be suitable for anything due to the fact that it is outdated or lacks security patches. But will be free 😄

**Comment by xlmnxp** [comment]
We need community fork of directus before license change,
that unfair to benefits from community contributions and marketing as open source project then close it source code

**Comment by benhaynes** [comment]
> We need community fork of Directus before license change, that unfair to benefits from community contributions and marketing as open source project then close it source code

Directus is **not** closed source under BSL... and all contributors of the platform assigned their copyrights to us so that we were able to make this change. Please read all of the discussions and posts regarding this change so that you're aware of these important details.

Directus 9 is still GPLv3, so you're welcome to fork and maintain it. However, in doing so, you'll quickly realized how much work is involved in that effort... and why we needed to make this change to sustain development.

**Comment by xlmnxp** [comment]
> > We need community fork of Directus before license change, that unfair to benefits from community contributions and marketing as open source project then close it source code
> 
> Directus is **not** closed source under BSL... and all contributors of the platform assigned their copyrights to us so that we were able to make this change. Please read all of the discussions and posts regarding this change so that you're aware of these important details.
> 
> Directus 9 is still GPLv3, so you're welcome to fork and maintain it. However, in doing so, you'll quickly realized how much work is involved in that effort... and why we needed to make this change to sustain development.

the model you doing is closed because after 10 years the version of Directus will be unusable and have a lot of security bugs need to be patch

so If any company contribute feature now for Directus they cannot benefits from it until 10 years pass

**Comment by benhaynes** [comment]
> the model you doing is closed because after 10 years the version of Directus will be unusable and have a lot of security bugs need to be patch. so If any company contribute feature now for Directus they cannot benefits from it until 10 years pass

You're talking about Directus 9 (GPLv3) becoming stale without patches/updates. What I am saying is that Directus 10+ (BSL) is still open-source and free for nearly all users because we added very permissive usage grants to the license.

I hope that clears things up.

**Comment by louisdussarps** [comment]
Hello,
As fair as the directus license change is, it can have significant implications for projects that were built on Directus and may have been operating under different assumptions regarding the licensing terms.
In order to maintain a stable version of directus v9, we made this fork : https://github.com/directus9/directus9.
Our goal is mainly to fixes bug and security issues when they arise.
We hope this fork can help other teams and gather a community's support in maintaining Directus9

**Comment by xlmnxp** [comment]
> Hello, As fair as the directus license change is, it can have significant implications for projects that were built on Directus and may have been operating under different assumptions regarding the licensing terms. In order to maintain a stable version of directus v9, we made this fork : https://github.com/directus9/directus9. Our goal is mainly to fixes bug and security issues when they arise. We hope this fork can help other teams and gather a community's support in maintaining Directus9

I'm supporting you, maybe it will success like Amazon OpenSearch

**Comment by tomdavidson** [comment]
I understand this is old news but new to me. 

Directus and Directus Cloud has been on my radar for a while because we have a strong bias for managed Open Source. I finally have the project where it would be a perfect fit, but we treat BSL the same as other proprietary vendors due to the inconsistent encumbering restrictions and not being an OSI approved  license.  Directus no longer makes the cut. This is a bummer. Now its just yet another one of many proprietary apps this space.

We do not use MongoDB, Terraform, Kafka, ElasticSearch, etc for the same reasons and they all have much stronger value props.

**Pull request merged**